### PR TITLE
Adds the WT-551, Unskyrats the WT-550 ammo

### PR DESCRIPTION
**Pull request by BurgerLUA**
## The WT-551
image

This adds the WT-551. A remade version of the WT-550 that is worse in every way. Fortunately, that means that it is balanced enough to be put in NanoTrasen armories.

Compared tot he WT-551, it is bulkier and slightly slower (0.3 second fire delay compared to 0.2). Additionally, it is commonly used with rubber-tipped rounds or FlatHead rounds, which are a special surplus of ammo that deals less damage and has no wounding, embedding, or penetrative power. Regular ammo can be purchased from cargo or researched later, with special ammo also being available later.

Note that this does not replace the WT-550.

## FlatHead ammo

image

Flathead ammo deals 18 brute damage (compared to the original 20), and 5 stamina damage per hit. It is extremely weak against armor, has no embed chance, has virtually no wounding chance. It's perfect for cheap corporate companies dealing with cheaper personnel. This is the type of lethal ammo that security will use for the gun, unless someone speedruns weapon research.

## Research Progression

image

### Basic WT-550/WT-551 Ammunition.
Flathead rounds and Rubber rounds for the WT-550/WT-551 can be researched for 2500 points after unlocking the "Weapon Development Technology" node.

### Advanced WT-550/WT-551 Ammunition.
Regular rounds and AP rounds for the WT-550/WT-551 can be researched can be researched for 5000 points after unlocking the "Advanced Weapon Development Technology" and "Basic WT-550/WT-551 Ammunition" nodes.

### Illegal WT-550/WT-551 Ammunition.
Incendiary rounds for the WT-550/WT-551 can be researched can be researched for 7500 points after unlocking the "Illegal Technology", "Exotic Ammo" , and "Advanced WT-550/WT-551 Ammunition" nodes.

### Syndicate Research

Removes the WT-550 ammo from syndicate research since it is now redundant.

## Cargo

image

WT-551 rifles can be ordered in pairs (2) for the cost of a parrot, a grilling starter pack, or a crab rocket (1600 credits). This value was chosen because it is slightly higher than the thermal pistols, and the traitor-ordered WT-550 rifle pack (which contains lethal ammo + spare lethal ammo).

Additional FlatHead, Rubber, and Regular ammo can be ordered from cargo as well.

Cargo techs no longer get WT-550s in the mail, but instead WT-551s (why was this a thing holy shit).

## Armory

image

2 WT-551s can be found in the armory.
For balance purposes one (1) laser rifle was removed.

## I hate Skyrat so much holy shit

image

Unfucks the WT-550 ammo types by removing their dumb names and changed caliber types.

Unfucks the WT-550 ammo in the ammo printer so that rubber rounds can be printed at T0 and everything else (except incendiary rounds) can be printed with the adv munitions disk.

The bullets for the WT-550 have been forcibly changed to /tg/ balance, which means that any and all future Skyrat PRs cannot touch the damage values for it (unless some fuckery occurs, idk).

## Why It's Good For The Game

image

Security is in dire need of actual ballistics. /tg/ removed ballistics from security because of reasons I legitimately don't think are valid. It's also a huge balance concern for security not to have at least 1 ballistic weapon (other than the shotgun) because it doesn't stop antags from hoarding laser immunity or meds.

Also guns are cool.

## Changelog

:cl: BurgerBB
add: Adds the WT-551 rifle, a redesign of the WT-550 rifle that is balanced (citation needed) for security use.
add: Makes WT-550 ammo researchable and orderable from cargo. Removes WT-550 ammo from syndicate research, and gives them their own categories.
/:cl:

### CODE REVIEW

File: modular_zubbers/modules/ballstics/wt_autorifle.dm
Comment (StrangeWeirdKitten) [inline modular_zubbers/modules/ballstics/wt_autorifle.dm:146]: There's no reason to make three separate tech nodes. Just throw them into ``modular_zubbers\code\modules\research\techweb\all_nodes.dm`` under pre-existing tech webs with the ``/datum/techweb_node/New()`` proc.

**Review by StrangeWeirdKitten — CHANGES_REQUESTED** [review]

**Comment by StrangeWeirdKitten** [comment]
Other than technical stuff, another maintainer can go through this and sign off on it. But I don't personally see an issue with test merging something like this.

**Comment by YakumoChen** [comment]
It's also a huge balance concern for security not to have at least 1 ballistic weapon (other than the shotgun) because it doesn't stop antags from hoarding laser immunity or meds.

Antags should have a counter against security, otherwise its just going to be like on skyrat where security constantly equips ballistics so EMPs aren't worth anything

File: code/modules/research/techweb/all_nodes.dm
Comment (StrangeWeirdKitten) [inline code/modules/research/techweb/all_nodes.dm:2238]: Modularize this inside ``modular_zubbers\code\modules\research\techweb\all_nodes.dm``

File: modular_zubbers/modules/ballstics/wt_autorifle.dm
Comment (StrangeWeirdKitten) [inline modular_zubbers/modules/ballstics/wt_autorifle.dm:132]: Put this inside ``modular_zubbers\code\modules\research\techweb\all_nodes.dm`` I want all techweb New() procs located inside the same folder using ``design_ids += list()`` for consistency.

File: modular_zubbers/modules/ballstics/wt_autorifle.dm
Comment (StrangeWeirdKitten) [inline modular_zubbers/modules/ballstics/wt_autorifle.dm:52]: Typo

```suggestion
//AMMO CASING
```

**Review by StrangeWeirdKitten — CHANGES_REQUESTED** [review]
Just one more thing

**Comment by StrangeWeirdKitten** [comment]
Once these changes get implemented, I'll push for this PR to get test merged and we'll try it out.

File: modular_zubbers/modules/ballstics/wt_autorifle.dm
Comment (projectkepler-ru) [inline modular_zubbers/modules/ballstics/wt_autorifle.dm:37]:
```suggestion
	stamina = 16
```
could have it at 16 instead so it's a  7.5 hit till you get dropped from stamina damages

**Comment by projectkepler-ru** [comment]
I feel like I'd like it more if this was also not full auto- but I'll have to see this in a round first since shooting at non moving target isn't exactly giving me a lot of conclusive results

**Comment by BurgerLUA** [comment]
> I feel like I'd like it more if this was also not full auto- but I'll have to see this in a round first since shooting at non moving target isn't exactly giving me a lot of conclusive results

It has the same DPS as the laser gun when using flathead ammo. This DPS will actually be less when you consider that a lot of traitor gear has high brute resist.

**Comment by projectkepler-ru** [comment]
> > I feel like I'd like it more if this was also not full auto- but I'll have to see this in a round first since shooting at non moving target isn't exactly giving me a lot of conclusive results
> 
> It has the same DPS as the laser gun when using flathead ammo. This DPS will actually be less when you consider that a lot of traitor gear has high brute resist.

mmmm I see

File: modular_zubbers/modules/ballstics/wt_autorifle.dm
Comment (BurgerLUA) [reply modular_zubbers/modules/ballstics/wt_autorifle.dm:132]: can't you just do that yourself when the time comes so I don't have to deal with .dme conflicts when everyone uses that folder

File: code/modules/research/techweb/all_nodes.dm
Comment (BurgerLUA) [reply code/modules/research/techweb/all_nodes.dm:2238]: I personally don't think that's good practice because if they ever change those ids, there would be issues.

Comment (StrangeWeirdKitten) [reply code/modules/research/techweb/all_nodes.dm:2238]: It'll realistically be on my end to deal with. There's no 100 percent way to stop all conflicts. Throw it inside a modular proc, even if it's inside ``modular_zubbers/modules/ballstics/wt_autorifle.dm``. Historically on skyrat not modularizing techweb edits has caused way more conflicts than the occasional two-second edit if TG decides to ever change these.

File: modular_zubbers/modules/ballstics/wt_autorifle.dm
Comment (StrangeWeirdKitten) [reply modular_zubbers/modules/ballstics/wt_autorifle.dm:132]: Sure

**Comment by YakumoChen** [comment]
> This DPS will actually be less when you consider that a lot of traitor gear has high brute resist.

Ballistics use explicitly bullet resist though? Bullet resistance isn't as common or great

Overall it will be harder for antags and much easier for security to have ballistic options, since counters to ballistics are rarer and make you much more vulnerable to energy weapons

**Comment by BurgerLUA** [comment]
If the PR ends up destroying antags, the gun can be nerfed.

**Comment by StrangeWeirdKitten** [comment]
> If the PR ends up destroying antags, the gun can be nerfed.

I'm opening a feedback thread. [here](https://github.com/Bubberstation/Bubberstation/pull/655). All feedback / balance concerns need to go into this.

I'm going to test merge this PR. @BurgerLUA if it's successful, I'm still going to want modularization of the core TG all_nodes I asked for before a full merge. That can be done any time during the week or so I run this.

**Comment by StrangeWeirdKitten** [comment]
@BurgerLUA here's a oversite. You can just remove the flathead ammo from the mag and replace it inside the ammo lathe on basic tech.

image

**Comment by BurgerLUA** [comment]
> @BurgerLUA here's a oversite. You can just remove the flathead ammo from the mag and replace it inside the ammo lathe on basic tech.
> 
> image

That's a Skyrat issue.

**Comment by BurgerLUA** [comment]
> @BurgerLUA here's a oversite. You can just remove the flathead ammo from the mag and replace it inside the ammo lathe on basic tech.
> 
> image

fixed

File: code/modules/research/techweb/all_nodes.dm
Comment (BurgerLUA) [reply code/modules/research/techweb/all_nodes.dm]: fixed

File: modular_zubbers/modules/ballstics/wt_autorifle.dm
Comment (BurgerLUA) [reply modular_zubbers/modules/ballstics/wt_autorifle.dm]: I think 15 is fine for now unless people think it's too weak.

**Review by StrangeWeirdKitten — APPROVED** [review]

**Comment by Majkl-J** [comment]
Does this replace solfed police equipment too or just regular sec?

**Comment by BurgerLUA** [comment]
> Does this replace solfed police equipment too or just regular sec?

just regular sec im pretty sure

**Review by projectkepler-ru — APPROVED** [review]

**Comment by projectkepler-ru** [comment]
Further testing are still being done but thus far It's been going well

File: modular_zubbers/code/modules/projectiles/guns/ballistic/automatic.dm
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/projectiles/guns/ballistic/automatic.dm:6]: I think this need a more distinctive naming,

File: modular_zubbers/code/modules/projectiles/guns/ballistic/automatic.dm
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/projectiles/guns/ballistic/automatic.dm:7]: If you referenced that this is either a civilian export variant/  Sol Firearm Code Compliant (which would explain why this thing is still around versus the WT-550) I'd dig it

File: modular_zubbers/code/modules/projectiles/projectile/bullets/smg.dm
Comment (projectkepler-ru) [inline modular_zubbers/code/modules/projectiles/projectile/bullets/smg.dm:36]:
```suggestion
	stamina = 16
```

I still feel like the stamina damage could be buffed up to 16 instead of 15 since the gun fires fairly slowly

**Comment by projectkepler-ru** [comment]
I like the sprites but  I'd like it more if it was somehow longer and not just 32x32
As it would be more visually noticable that this is something like what the SL8 is to the G36

**Comment by BurgerLUA** [comment]
> I like the sprites but I'd like it more if it was somehow longer and not just 32x32 As it would be more visually noticable that this is something like what the SL8 is to the G36

Making a gun that is outside the 32x32 size is too much effort to get right. The sprite is fine.

**Comment by projectkepler-ru** [comment]
> > I like the sprites but I'd like it more if it was somehow longer and not just 32x32 As it would be more visually noticable that this is something like what the SL8 is to the G36
> 
> Making a gun that is outside the 32x32 size is too much effort to get right. The sprite is fine.

there's also some other thing I said but yeah

**Comment by BurgerLUA** [comment]
The sprite is fine. It looks like a WT-551 without being too crazy while still being distinct.

**Comment by StrangeWeirdKitten** [comment]
Sprites and name edits can come later in another PR or other fluff things if someone is interested. I'm going to merge this once CI gives the final green light

**Comment by Cyprex** [comment]
Should we just remove the ammo lathe?

**Pull request merged**
